rfcs/0000-cluster-signing.md

```diff
@@ -0,0 +1,109 @@
+
+Secret format example:
+```yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cosign-key
+  annotations:
+    kpack.io/cosign-credentials: ""
+    # Optional
+    kpack.io/cosign.repository: "<repository-url>"
+    # Optional
+    kpack.io/cosign.docker-media-types: "1"
+type: Opaque
+stringData:
+  cosign.key: <cosign-private-key>
+  cosign.pub: <cosign-public-key>
+  cosign.password: <cosign-private-key-passphrase>
+```
+
+### Generating the signature payload
+
+In order to generate the signature payload on the controller-side `kpack` will do the following for every build - 
+
+- Use the image digest output from the `Build` from the user-namespace pod. The controller should have the appropriate RBAC permissions to read this and to verify that this is not tampered with.
+- Use this digest to create a `cosign` payload on the controller side for each `cosign` secret attached to the `kpack` controller service account.
+- Use the appropriate `docker-registry` secrets attached to the kpack controller service account to push this signed payload to an appropriate registry.
+- Update the `Build` and `Image` status to reflect the output from pushing the signature.
+
+This assumes that the `kpack` controller service account is configured appropriately to have the `cosign` and [`docker-registry`](https://github.com/pivotal/kpack/blob/main/docs/secrets.md#docker-registry-secrets) secrets as needed.
+
+If the controller service account has no `cosign` secrets, it will not attempt to do any cluster-wide image signing. If the controller has a `cosign` service account secret but doesn't have corresponding `docker-registry` secrets, it should fail with an appropriate status and attach a scrubbed reason the the user's `Image`/`Build` output.
+
+### Authentication to a container registry
+
+`cosign` uses the `DefaultKeychain` from [`go-containerregistry`](https://github.com/google/go-containerregistry/blob/main/pkg/authn/README.md#tldr-for-consumers-of-this-package) to authenticate the command-line interface using the machine's Docker CLI
+credentials. For `kpack`, it may be possible to use the same authentication mechanism that is in-place for uploading an image.
+
+## Complexity
+
+Integration with `cosign` is estimated to be of medium complexity. Since `cosign` is also written in Go, it is possible to import code from the tool directly within `kpack`, which can help make the integration easier.
+
+## Prior art
+
+- [Issue](https://github.com/pivotal/kpack/issues/684) filed on May 4th.
+- [cosign RFC](https://github.com/pivotal/kpack/pull/694)
+- [Use of `cosign` in Kubernetes](https://github.com/kubernetes/release/pull/2016)
+  for validation of distroless images.
+- [Use of `cosign` in Connaisseur](https://github.com/sse-secure-systems/connaisseur/pull/107)
+  for policy enforcement in container images.
+- [Notary v1 implementation pull request](https://github.com/pivotal/kpack/pull/541).
+
+## Alternatives
```

Comment:
Could this be a completely separate kubernetes deployment that automatically signs every `status.latestImage` produced by kpack?